Writing-editing network/main.py

```diff
@@ -104,30 +121,141 @@ def _mask(prev_generated_seq):
         mask = mask.cuda()
     return prev_generated_seq.data.masked_fill_(mask, 0)
 
-def train_batch(input_variable, input_lengths, target_variable, topics, model,
-                teacher_forcing_ratio):
+def load_training_samples_for_shuffling(dataset):
+    train_loader = DataLoader(dataset, config.batch_size)
+    for d in train_loader:
+        train_shuffle_samples.append(d)
+
+def freeze_generator():
+    for param in model.parameters():
+        param.requires_grad = False
+
+def unfreeze_generator():
+    for param in model.parameters():
+        param.requires_grad = True
+
+def freeze_discriminator():
+    for param in discrim_model.parameters():
+        param.requires_grad = False
+
+def unfreeze_discriminator():
+    for param in discrim_model.parameters():
+        param.requires_grad = True
+
+
+def train_discriminator(input_variable, target_variable, is_eval=False):
+    sequence_length = input_variable.shape[1]
+    '''add other return values'''
+    dis_out, dis_sig = discrim_model(input_variable, sequence_length, config.batch_size)
```

Comment:
Wrong dimensions are coming here. Kindly check. The dimensions of `dis_out` and `target_variable` should match. The target variable has a dimensionality of `(20, 1)`